### ISSUE TITLE
Implement Provenance

### ISSUE THREAD

**Issue by Veldhoen** [CLARIAH/DANE#19]
In library, create provenance module and/or add as object to models

**Comment by Veldhoen** [comment]
And reference wheel (local build) (or later: Github url) in pyproject
(In the end: push to Pypi)

**Comment by jblom** [comment]
**Update**: currently will work on fixing the provenance chain in both VisXP workers. Now it's a bit hard to follow and not entirely correct. Probably will also put the common provenance functions in the DANE lib

**Comment by jblom** [comment]
added the new provenance module to DANE 0.4.2 https://github.com/CLARIAH/DANE/blob/main/dane/provenance.py

applied it to the 1st worker in this draft PR: https://github.com/beeldengeluid/dane-video-segmentation-worker/pull/36/files

It's mostly ok, some rough edges need to be polished
